src/main/java/com/github/maleksandrowicz93/cqrsdemo/student/UpdatePasswordCommandHandler.java

```diff
@@ -0,0 +1,42 @@
+package com.github.maleksandrowicz93.cqrsdemo.student;
+
+import com.github.maleksandrowicz93.cqrsdemo.student.exception.InvalidCredentialsException;
+import com.github.maleksandrowicz93.cqrsdemo.student.exception.PasswordNotUpdatedException;
+import com.github.maleksandrowicz93.cqrsdemo.student.exception.StudentNotFoundException;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Component;
+
+import java.util.UUID;
+
+@Log4j2
+@Component
+@RequiredArgsConstructor
+class UpdatePasswordCommandHandler {
+
+    final StudentRepository studentRepository;
+    final PasswordEncoder passwordEncoder;
+
+    void handle(UUID studentId, String password) {
+        if (StringUtils.isBlank(password)) {
+            log.error("Invalid password. Cannot be updated for student with id: {}.", studentId);
+            throw new InvalidCredentialsException();
+        }
+        var student = studentRepository.findById(studentId)
+                .orElseThrow(() -> {
+                    log.error("Not found student with id: {}", studentId);
+                    throw new StudentNotFoundException();
+                });
+        if (password.equals(student.password())) {
+            throw new PasswordNotUpdatedException();
```

Comment:
Nie wiem czy to aż takie wyjątkowo straszne, że hasło obecnie jest takie samo jak docelowe, żeby aż wyjątek rzucać. Celem tej metody jest zmiana hasła na X - jeśli hasło po jej zakończeniu jest X, to raczej nieistotne, że przed też było X, a jeśli istotne, to raczej na informację dodatkową, pomimo poprawnego zakończenia przetwarzania. To jest też kolejny problem z Exception Driven Architecture - "wyjątkowość" traci znaczenie. Nie mówiąc o overheadzie z obsługi wyjątków latających wszędzie.